### ISSUE TITLE
A better way to automate spell point use, than my previous suggestion

### ISSUE THREAD

**Issue by Ronnie-McDurgen** [kakaroto/Beyond20#728]
Here is a much better idea for automating spell point usage than I mentioned in my previous feature suggestion. Or at least I personally think that this one is much better. It is up to you if you want to close the old feature request, or leave it open.

Keep in mind that this feature will be able to reuse code from the feature that lets the user define a chat message that is dumped into the roll20 chat whenever a spell is used... because this feature would do _the exact same thing_, except based on the level of the spell that was just cast.

Anyway, without further explanation, here is the description for my feature suggestion.

### Full Description:

**1)** The user can go to the Beyond20 settings and click on a button that says "Automate Spell Point Usage (roll20 only)", or "Use Mana Instead of Spell Slots (roll20 only)". Or some other option, that you like better than these two.

**2)** Clicking that button makes a new settings window open up.

**3)** In that window, the following requests to the user will appear:

"Please enter the name of the macro that you want to use whenever a level 1 spell is used"
And below that line of text, there is a textbox that the user can type into.

"Please enter the name of the macro that you want to use whenever a level 2 spell is used"
And below that line of text, there is a textbox that the user can type into.

Etc...

**4)** Then, whenever this user casts a spell, the correct macro will run.

### The End

And again, thank you so much for creating this wonderful extension!

**Comment by Aeristoka** [comment]
The purpose of Beyond20 has been and continues to be to use DnDBeyond as the "Source of Truth" to roll to various VTTs, and not to spend things on the other side. This request much more specifically requests specific Roll20-only code to make things happen.

I'll leave it to @kakaroto to rule on this, but I am in doubt this would happen.

**Comment by Ronnie-McDurgen** [comment]
> The purpose of Beyond20 has been and continues to be to use DnDBeyond as the "Source of Truth" to roll to various VTTs, and not to spend things on the other side. This request much more specifically requests specific Roll20-only code to make things happen.
> 
> I'll leave it to @kakaroto to rule on this, but I am in doubt this would happen.

Fair enough, no problem... I can always continue to use Beyond20's "add a text comment" feature to make Beyond20 run a roll20 macro for me :-)

However, the other feature that I mentioned (the "always show a spell's level" feature)... it would provide a lot of useful information to a lot of people, in a lot of different VTTs. :-)

For example, if I always had the spell's level in front of my eyes when I'm GMing, then it would be much easier for me to manually correct a mistake, when the user accidentally messes up his spellcasting somehow :-)

**Comment by Ronnie-McDurgen** [comment]
P.S. I have modified the other post, to make the request (hopefully) more relevant to Beyond20's mission statement :-)
